plugins/packages/oracledb/lib/index.ts

```diff
@@ -0,0 +1,164 @@
+import { SourceOptions, QueryOptions } from "./types";
+
+export default class OracledbQueryService implements QueryService {
+  private static _instance: OracledbQueryService;
+
+  constructor() {
+    if (OracledbQueryService._instance) {
+      return OracledbQueryService._instance;
+    }
+
+    OracledbQueryService._instance = this;
+    return OracledbQueryService._instance;
+  }
+
+  async run(
+    sourceOptions: SourceOptions,
+    queryOptions: QueryOptions,
+    dataSourceId: string,
+    dataSourceUpdatedAt: string
+  ): Promise<QueryResult> {
+    let result = {
+      rows: [],
+    };
+    let query = "";
+
+    if (queryOptions.mode === "gui") {
+      if (queryOptions.operation === "bulk_update_pkey") {
+        query = await this.buildBulkUpdateQuery(queryOptions);
+      }
+    } else {
+      query = queryOptions.query;
+    }
+
+    const knexInstance = await this.getConnection(
+      sourceOptions,
+      {},
+      true,
+      dataSourceId,
+      dataSourceUpdatedAt
+    );
+
+    try {
+      result = await knexInstance.raw(query);
+
+      return {
+        status: "ok",
+        data: result,
+      };
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async testConnection(
+    sourceOptions: SourceOptions
+  ): Promise<ConnectionTestResult> {
+    const knexInstance = await this.getConnection(sourceOptions, {}, false);
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const result = await knexInstance.raw("SELECT * FROM v$version");
+
+    return {
+      status: "ok",
+    };
+  }
+
+  initOracleClient(clientPathType:string, customPath:string) {
+    try {
+      if(clientPathType === 'custom'){
+        if (process.platform === "darwin") {
+          oracledb.initOracleClient({libDir: process.env.HOME + customPath});
+        } else if (process.platform === "win32") {
+          oracledb.initOracleClient({
+            libDir: customPath,
+          }); // note the double backslashes
+        }
+      }else{
+        oracledb.initOracleClient(); 
+      }
+    } catch (err) {
+      console.error(err);
+      throw err;
+    }
+  }
+
+  async buildConnection(sourceOptions: SourceOptions) {
+    // we should add this to our datasource documentation
+    try{
+      oracledb.oracleClientVersion
+    }catch(err){
+      console.log('Oracle client is not initailized');
+      this.initOracleClient(sourceOptions.client_path_type, sourceOptions.path)
+    }
+
+    const config: Knex.Config = {
+      client: "oracledb",
+      connection: {
+        user: sourceOptions.username,
+        password: sourceOptions.password,
+        connectString: `(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=${sourceOptions.host})(PORT=${sourceOptions.port}))(CONNECT_DATA=(SERVER=DEDICATED)(${sourceOptions.database_type}=${sourceOptions.database})))`,
+        multipleStatements: true,
+        ssl: sourceOptions.ssl_enabled ?? false, // Disabling by default for backward compatibility
```

Comment:
Since this is newly introduced datasource, we can simply keep `sourceOptions.ssl_enabled`